src/dwarf.cpp

```diff
@@ -1019,59 +1088,58 @@ void dwarf::implementation::process() {
                 path_identifier_push();
             }
 
+            if (attributes.has(dw::at::type)) {
+                attributes.get(dw::at::type)._value.string(resolve_type(attributes.get(dw::at::type)));
+            }
+
+            die._skippable = skip_die(die, attributes);
             die._ancestry = _ancestry;
-            die._hash = die_hash(die); // precompute the hash we'll use for the die map.
+            die._hash = die_hash(die, attributes);
+            die._fatal_attribute_hash = fatal_attribute_hash(attributes);
 
-            dies.push_back(std::move(die));
+            dies.emplace_back(std::move(die));
         }
     }
 
-    // Now that we have all the dies for this translation unit, do some post-processing to get more
-    // details out and reported to the surface.
-
     dies.shrink_to_fit();
 
-    for (auto& d : dies) {
-        resolve_reference_attributes(dies, d);
-
-        resolve_type_attribute(dies, d);
-#if 0
-        bool modified = resolve_specification_attribute(dies, d, _callbacks._empool);
-
-        if (modified) {
-            //
-            // Redo some finalization of the DIE
-            //
-            path_identifier_set(die_identifier(d));
+    _register_dies(std::move(dies));
+}
 
-            d._path = empool(std::string_view(qualified_symbol_name(d)));
+/**************************************************************************************************/
 
-            // recompute the die hash
-            // REVISIT (fosterbrereton) : There are a number of values that,
-            // when changed, require a re-hash of the die (arch, tag, and path). The setting of those values
-            // so either be tucked within an API of the die (so it can rehash the value
-            // on its own) or done at the end of these resolve_*
-            d._hash = die_hash(d); // precompute the hash we'll use for the die map.
-        }
-#endif
+die_pair dwarf::implementation::fetch_one_die(std::size_t debug_info_offset) {
+    if (!_ready && !register_sections_done()) throw std::runtime_error("dwarf setup failed");
+    assert(_ready);
```

Comment:
The `throw` then `assert()` reads oddly. Again, not sure if wrong, but shouldn't it all be throw or assert?

Reply:
`register_sections_done` affects `_ready`. So if `_ready` is false, `register_sections_done` is called which should set `_ready` to true. If _that_ fails, we throw. Since the exception is being thrown there, yeah, the assert isn't necessary.